src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -288,3 +288,31 @@ void WaveformWidgetRenderer::setTrack(TrackPointer track) {
         m_rendererStack[i]->onSetTrack();
     }
 }
+
+std::optional<CuePointer> WaveformWidgetRenderer::getCueAtPoint(QPoint point) {
```

Comment:
Is wrapping a pointer into std::optional common practice in C++? I guess in this case returning a nullptr is appropriate and simplifies the code. Checking for both nullopt and nullptr feels overcomplicated.